syntax = "proto3";

package mqtt;

service MqttService {
  rpc PublishDeviceSettings (PublishDeviceSettingsRequest) returns (PublishDeviceSettingsResponse) {}
}

message Settings {
  string room_id = 1;
  string device_id = 2;
  float temperature = 3;
  uint32 fan_speed = 4; // 0-100
}

message PublishDeviceSettingsRequest {
  Settings settings = 1;
}

message PublishDeviceSettingsResponse {
  string message = 1;
}
